src/main/java/io/confluent/connect/jdbc/util/CachedConnectionProvider.java

```diff
@@ -53,6 +53,7 @@ public synchronized Connection getValidConnection() {
       } else if (!connection.isValid(VALIDITY_CHECK_TIMEOUT_S)) {
         log.info("The database connection is invalid. Reconnecting...");
         closeQuietly();
+        connection = null;
```

Comment:
Isn't this redundant since `closeQuietly()` already does this? If the issue is that there is an exception during the call to `connection.close()` in that method, then perhaps the `connection = null;` line should simply be moved to a `finally` clause in that method?